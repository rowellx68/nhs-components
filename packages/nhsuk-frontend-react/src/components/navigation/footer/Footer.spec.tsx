import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  FooterWithKeyLinksOnly as FooterWithKeyLinksOnlyStory,
  FooterWithNavigationLinks as FooterWithMetaLinksOnlyStory,
} from './Footer.stories';

const FooterWithKeyLinksOnly = composeStory(FooterWithKeyLinksOnlyStory, meta);
const FooterWithMetaLinksOnly = composeStory(
  FooterWithMetaLinksOnlyStory,
  meta,
);

it('should render the footer component', () => {
  const { container } = render(<FooterWithKeyLinksOnly />);

  expect(container.querySelectorAll('a')).toHaveLength(5);
  expect(container.querySelector('footer')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the footer component', () => {
  const { container } = render(<FooterWithMetaLinksOnly />);

  expect(container.querySelectorAll('a')).toHaveLength(21);
  expect(container.querySelector('footer')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
