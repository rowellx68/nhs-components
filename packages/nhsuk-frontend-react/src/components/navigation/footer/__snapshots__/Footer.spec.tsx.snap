// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the footer component 1`] = `
<div>
  <footer
    role="contentinfo"
  >
    <div
      class="nhsuk-footer-container"
    >
      <div
        class="nhsuk-width-container"
      >
        <h2
          class="nhsuk-u-visually-hidden"
        >
          Support links
        </h2>
        <div
          class="nhsuk-footer"
        >
          <ul
            class="nhsuk-footer__list"
          >
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Accessibility statement
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Contact us
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Cookies
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Privacy policy
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Terms and conditions
              </a>
            </li>
          </ul>
          <div>
            <p
              class="nhsuk-footer__copyright"
            >
              © 
              NHS England
            </p>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`should render the footer component 2`] = `
<div>
  <footer
    role="contentinfo"
  >
    <div
      class="nhsuk-footer-container"
    >
      <div
        class="nhsuk-width-container"
      >
        <h2
          class="nhsuk-u-visually-hidden"
        >
          Support links
        </h2>
        <div
          class="nhsuk-footer"
        >
          <ul
            class="nhsuk-footer__list"
          >
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Home
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Health A to Z
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Live Well
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Mental health
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Care and support
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Accessibility statement
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Pregnancy
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                NHS services
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Coronavirus (COVID-19)
              </a>
            </li>
          </ul>
          <ul
            class="nhsuk-footer__list"
          >
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                NHS App
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Find my NHS number
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Your health records
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                About the NHS
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Healthcare abroad
              </a>
            </li>
          </ul>
          <ul
            class="nhsuk-footer__list"
          >
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Contact us
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Other NHS websites
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Profile editor login
              </a>
            </li>
          </ul>
          <ul
            class="nhsuk-footer__list nhsuk-footer__meta"
          >
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                About us
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Accessibility statement
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Our policies
              </a>
            </li>
            <li
              class="nhsuk-footer__list-item nhsuk-footer-default__list-item"
            >
              <a
                class="nhsuk-footer__list-item-link"
                href="#"
              >
                Cookies
              </a>
            </li>
          </ul>
        </div>
        <div>
          <p
            class="nhsuk-footer__copyright"
          >
            © 
            NHS England
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
